test/cmdlineTests.sh

```diff
@@ -334,16 +330,16 @@ function test_via_ir_equivalence()
 
     msg_on_error --no-stderr "$SOLC" --ir-optimized --debug-info location "${optimizer_flags[@]}" "$solidity_file" |
         sed '/^Optimized IR:$/d' |
-        split_on_empty_lines_into_numbered_files $output_file_prefix ".yul"
-
-    for yul_file in $(find . -name "${output_file_prefix}*.yul" | sort -V); do
-        msg_on_error --no-stderr "$SOLC" --strict-assembly --asm "${optimizer_flags[@]}" "$yul_file" |
-            sed '/^Text representation:$/d' > "${yul_file/.yul/.asm}"
-    done
+        split_on_empty_lines_into_numbered_files "$output_file_prefix" ".yul"
 
     local asm_output_two_stage asm_output_via_ir
-    for asm_file in $(find . -name "${output_file_prefix}*.asm" | sort -V); do
-        asm_output_two_stage+=$(sed '/^asm_output_two_stage:$/d' "$asm_file" | sed '/^=======/d')
```

Comment:
Why was this removed? Is it still make sense to call it `asm_output_two_stage`? Now it is only one, no?

Reply:
No, it's still two-stage. This bit was just removing some lines from the `--asm` output, not doing compilation:
1) The `=======` separators. This is still happening. See the loop immediately below.
2) Lines containing `asm_output_two_stage:`. Not sure why actually since the output is not supposed to contain anything like this. Could be a left-over from how this test looked earlier. In any case now this is superfluous so I removed it.

Overall, two loops are not necessary here. It can easily be done in one, just like it's already done for `--bin` output below.